pkg/controller/bundlec/controller.go

```diff
@@ -125,3 +194,77 @@ func (c *controllerIndexAdapter) ControllerByObject(gk schema.GroupKind, namespa
 	}
 	return objs, nil
 }
+
+func byConfigMapNamespaceNameIndex(obj interface{}) ([]string, error) {
+	d := obj.(*apps_v1.Deployment)
+	index := configMapNamespaceNameIndexKeysForContainers(d.Namespace, d.Spec.Template.Spec.Containers)
+	index = append(index, configMapNamespaceNameIndexKeysForContainers(d.Namespace, d.Spec.Template.Spec.InitContainers)...)
+	return index, nil
+}
+
+func configMapNamespaceNameIndexKeysForContainers(namespace string, containers []core_v1.Container) []string {
+	var indexKeys []string
+	for _, container := range containers {
+		for _, envFrom := range container.EnvFrom {
+			configMapRef := envFrom.ConfigMapRef
+			if configMapRef == nil {
+				continue
+			}
+			indexKeys = append(indexKeys, byConfigMapNamespaceNameIndexKey(namespace, configMapRef.Name))
+		}
+		for _, env := range container.Env {
+			valueFrom := env.ValueFrom
+			if valueFrom == nil {
+				continue
+			}
+
+			configMapKeyRef := valueFrom.ConfigMapKeyRef
+			if configMapKeyRef == nil {
+				continue
+			}
+			indexKeys = append(indexKeys, byConfigMapNamespaceNameIndexKey(namespace, configMapKeyRef.Name))
+		}
+	}
+	return indexKeys
+}
+
+func byConfigMapNamespaceNameIndexKey(configMapNamespace, configMapName string) string {
+	return configMapNamespace + "/" + configMapName
```

Comment:
Any specific reason you use this style over `fmt.Sprintf`?

Reply:
It is not important in this case of course, but I think concatenation is faster - don't need to parse the formatting string. Also this code does not depend on the `fmt` package that way (also not important).